File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int reqCount = n/2;
        Map<Integer, Integer> myMap = new HashMap<>();
        for(int i = 0; i < n; i++) {
            myMap.put(nums[i], myMap.getOrDefault(nums[i], 0)+1);
            if(myMap.get(nums[i]) > reqCount) {
                return nums[i];
            }
        }
        return -1;
    }
}

File: 169-majority-element/majority-element.go
func majorityElement(nums []int) int {
    ans := nums[0]
    cnt := 1
    for i := 0; i < len(nums); i++{
        if(nums[i] == ans){
            cnt++
        }else{
            cnt--
        }

        if(cnt == 0){
            ans = nums[i]
            cnt = 1
        }
    } 
    return ans  
}